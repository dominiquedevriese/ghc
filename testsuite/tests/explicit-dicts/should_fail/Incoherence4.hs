module Incoherence4 where

class A a

class A a => B a

class B a => C a

foo :: (A a, C a) => a
foo = undefined

bar :: A a => C.Dict a -> a
bar dict = foo ((dict :: C a))
