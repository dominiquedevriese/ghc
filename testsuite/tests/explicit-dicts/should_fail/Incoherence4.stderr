
Incoherence4.hs:13:12: error: [-Wincoherence (in -Wdefault), -Werror=incoherence]
    • Explicit dictionary application to: ‘C a’
      is not allowed because of possible incoherence:
        Through an ancestor: ‘A a’ occurs in the context
    • In the expression: foo ((dict :: C a))
      In an equation for ‘bar’: bar dict = foo ((dict :: C a))
